Remotion/Security/Core/ThreadPrincipalProvider.cs

```diff
@@ -33,9 +33,9 @@ public ThreadPrincipalProvider ()
 
     public ISecurityPrincipal GetPrincipal ()
     {
-      // TODO RM-7872: Add notnull assertions
-      IIdentity identity = Thread.CurrentPrincipal!.Identity!;
-      if (!identity.IsAuthenticated)
+      var identity = Thread.CurrentPrincipal?.Identity;
+
+      if (identity is null or { IsAuthenticated: false })
```

Comment:
Exit-Condition code style preference: https://re-motion.atlassian.net/browse/RMTEAM-453

Reply:
@MichaelKetting  should I change it now or do we change it later?

Reply:
As you like. There's certainly other places in the code base and and I'm not too concerned with the odd null case since the compiler also helps.  It just served as a reminder that there's an analyzer ticket to create :) If you think it's a quick win, we can schedule the ticket in one of the next sprints.